hydromt/raster.py

```diff
@@ -2364,8 +2366,15 @@ def to_xyz_tiles(
                 shape, dims = temp.raster.shape, temp.raster.dims
                 pad_sizes = [tile_size - s for s in shape]
                 if any((s > 0 for s in pad_sizes)):
-                    pad_sizes = {d: (0, s) for d, s in zip(dims, pad_sizes) if s > 0}
-                    temp = temp.pad(pad_sizes, mode="constant", constant_values=nodata)
+                    pad_dict = {d: (0, s) for d, s in zip(dims, pad_sizes) if s > 0}
+                    temp = temp.pad(pad_dict, mode="constant", constant_values=nodata)
+                    # pad nan coordinates; this is not done by xarray
+                    for d, s in zip(dims, pad_sizes):
+                        if s > 0:
+                            coords = temp[d].values
+                            res = dst_res if d == x_dim else -dst_res
```

Comment:
can resolutions be negative? this seems odd to me, but perhaps I'm not quite understanding what's happening here.

Reply:
Yes, the orientation of the grid is also captured in the res. A grid which is is oriented north > south has a negative y res